platform-server-kernel/src/main/java/uno/cod/platform/server/core/domain/Challenge.java

```diff
@@ -83,4 +84,15 @@ public ZonedDateTime getEndDate() {
     public void setEndDate(ZonedDateTime endDate) {
         this.endDate = endDate;
     }
+
+    public void addTask(Task task) {
+        if (task == null) {
+            throw new IllegalArgumentException("task not valid");
```

Comment:
If you would throw that in a controller or maybe a mapper I would understand, but that's a domain class ...